image/templates/helm/shared/templates/_scanner-v4_init.tpl.htpl

```diff
@@ -0,0 +1,135 @@
+
+
+{{/* These will be propagated up. */}}
+{{ $components := dict "indexer" false "matcher" false }}
+{{ $dbEnabled  := false }}
+
+{{/* Handle high-level defaulting logic. */}}
+{{- if not $scannerV4Cfg.disable }}
+
+{{/* Sanity check. */}}
+{{- if not (or (eq $.Chart.Name "stackrox-central-services") (eq $.Chart.Name "stackrox-secured-cluster-services")) -}}
+  {{- include "srox.fail" (printf "Unexpected Helm chart name %q." $.Chart.Name) -}}
+{{- end -}}
+
+{{- if and (kindIs "invalid" $scannerV4Cfg.indexer.disable) (kindIs "invalid" $scannerV4Cfg.matcher.disable) -}}
+
+  {{/* Both settings scannerV4.indexer.disable and scannerV4.matcher.disable are not set by the user
+       => Pick default configuration based on the chart. */}}
+  {{- if eq $.Chart.Name "stackrox-central-services" -}}
+    {{- $components = $componentsCentralChart -}}
+  {{- else -}}
+    {{- $components = $componentsSecuredClusterChart -}}
+  {{- end -}}
+
+{{- else -}}
+
+  {{/* The user has explicitly set either scannerV4.indexer.disable and/or scannerV4.matcher.disable.
+       Therefore, we use this explicitly set configuration. */}}
+  {{- if not (kindIs "invalid" $scannerV4Cfg.indexer.disable) -}}
+    {{- $_ = set $components "indexer" (not $scannerV4Cfg.indexer.disable) -}}
+  {{- end -}}
+  {{- if not (kindIs "invalid" $scannerV4Cfg.matcher.disable) -}}
+    {{- $_ = set $components "matcher" (not $scannerV4Cfg.matcher.disable) -}}
+  {{- end -}}
+
+  {{/* This is potentially an unsupported configuration. Hence we generate a warning. */}}
+  {{- $msg := printf "indexer is %s and matcher is %s." (ternary "enabled" "disabled" (get $components "indexer")) (ternary "enabled" "disabled" (get $components "matcher")) -}}
+  {{- if eq $.Chart.Name "stackrox-central-services" -}}
+    {{- if or (ne (get $components "indexer") (get $componentsCentralChart "indexer")) (ne (get $components "matcher") (get $componentsCentralChart "matcher")) -}}
+      {{- include "srox.warn" (list $ (printf "Unsupported Scanner V4 configuration detected: %s" $msg)) -}}
+    {{- end -}}
+  {{- else -}}
+    {{- if or (ne (get $components "indexer") (get $componentsSecuredClusterChart "indexer")) (ne (get $components "matcher") (get $componentsSecuredClusterChart "matcher")) -}}
+      {{- include "srox.warn" (list $ (printf "Unsupported Scanner V4 configuration detected: %s" $msg)) -}}
+    {{- end -}}
+  {{- end -}}
+
+{{- end -}}
+
+{{/* Configure images, certificates and passwords as required. */}}
+{{/* Note that for secured-cluster-services we don't configure certificates here, instead they will be distributed at runtime by Sensor and Central. */}}
+{{ if get $components "indexer" }}
+  {{ include "srox.configureImage" (list $ $scannerV4Cfg.indexer.image) }}
+  {{- if eq $.Chart.Name "stackrox-central-services" -}}
+    {{/* Only generate certificate when installing central-services. */}}
+    {{ $cryptoSpec := dict "CN" "SCANNER_SERVICE: Scanner V4 Indexer" "dnsBase" "scanner-v4" }}
```

Comment:
I noticed that for the old scanner in case of the operator, the certificate are generated through operator code [here](https://github.com/stackrox/stackrox/blob/75681d95449249e1ca61fe12afd1f8135a41c643/operator/pkg/central/extensions/reconcile_tls.go#L71).

I don't know why that is but we should check if we also need to include options here to disable the certificate generation through helm in that case.

Reply:
Looking at the old scanner it also has no logic to deactivate that part: https://github.com/stackrox/stackrox/blob/305849f08f055e1c7413a7f3faaebde0a5ff5761/image/templates/helm/shared/templates/_scanner_init.tpl.htpl#L26

I don't really understand how it's ensured that there is no conflict between operator / helm chart certificate generation.

Reply:
When the chart is run by the operator, all helm cert generation logic is disabled.

Reply:
See https://github.com/stackrox/stackrox/blob/master/operator/pkg/central/values/translation/base-values.yaml and its securedcluster counterpart.

Reply:
Perfect, thank you @porridge that was the part I was missing.

Reply:
I mean, operator selectively sets `*erviceTLS.generate: false`. If scanner v4 introduces a new field, we will need to add such setting on operator side.